app/plugin_api.go

```diff
@@ -83,6 +83,10 @@ func (api *PluginAPI) SaveConfig(config *model.Config) *model.AppError {
 	return api.app.SaveConfig(config, true)
 }
 
+func (api *PluginAPI) GetServerVersion() string {
+	return model.CurrentVersion
```

Comment:
On the surface, I think there's nothing wrong with this: it returns the `CurrentVersion` as defined by the server back to the plugin for use. (Directly using `model.CurrentVersion` in the plugin would, of course, only return the version of the server against which the plugin was compiled.)

However, I'm worried about some of the helper methods in `model/version.go`. It's pretty natural to import this package (again, from the version you compiled against), but then things might start to act "weird". For example, `model.IsCurrentVersion` would return false if the server is newer. Technically, that's quite correct, but may not be obvious. And `model.IsPreviousVersionsSupported` might be weird?

I raise this only to inquire about how the server version is going to be used and whether or not we should have a story around using this value to avoid confusion. One idea might be to return not just the version here, but a struct containing the whole `model.versions` array, and exposing the same methods as the package on the struct.

Reply:
Lets break this up into multiple scenarios:
### Plugins `model` version is newer or the same as the servers one
I don't see how this can cause problem. The plugin has more "context" about the returned version and can handle it properly.

### Plugins `model` version is older as the servers one
Then unexpected thing can happen. But I don't see a use case where this can happen.

## Use cases
The use case of API methode I had in mind and also @seansackowitz [pointed out](https://pre-release.mattermost.com/core/pl/i4zw5mzm83yrxpjo78stj8cn1o) is the following: A plugins tries to be backwards compatible to older server versions. Because of this, it enables some functionaries e.g. bot accounts just to server how support them. To make this happen it need to get the current server version. 

A other use case I had in mind was a plugins which disallows older version of the server. If e.g.a change in `PostActionIntegrationRequest` is made in 5.4 the plugins wants to disallows older versions then 5.4. To do this, it need the current server version.

I don't see how a plugins could use the methods from https://github.com/mattermost/mattermost-server/blob/master/model/version.go. Check if a server is up top date seams a bit of a bad idea.